File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/exchange/HitBTCApiClient.java
package ru.nikitazhelonkin.coinbalance.data.api.exchange;


import java.util.HashMap;

import io.reactivex.Single;
import okhttp3.Credentials;
import ru.nikitazhelonkin.coinbalance.data.api.HttpErrorTransformer;
import ru.nikitazhelonkin.coinbalance.data.api.response.HitBTCBalancesResponse;
import ru.nikitazhelonkin.coinbalance.data.api.exchange.HitBTCApiService;

public class HitBTCApiClient implements ExchangeApiClient {

    private HitBTCApiService mApiService;

    public HitBTCApiClient(HitBTCApiService apiService){
        mApiService = apiService;
    }

    @Override
    public Single<HashMap<String, Float>> getBalances(String apiKey, String apiSecret) {
        return mApiService.balances( Credentials.basic(apiKey, apiSecret))
                .map(HitBTCBalancesResponse::getNonZeroBalances)
                .compose(new HttpErrorTransformer<>());
    }
}

File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/exchange/HitBTCApiClient.kt
package ru.nikitazhelonkin.coinbalance.data.api.exchange


import io.reactivex.Single
import okhttp3.Credentials
import ru.nikitazhelonkin.coinbalance.data.api.HttpErrorTransformer
import java.util.*

class HitBTCApiClient(private val mApiService: HitBTCApiService) : ExchangeApiClient {

    override fun getBalances(apiKey: String, apiSecret: String): Single<HashMap<String, Float>> {
        return mApiService.balances(Credentials.basic(apiKey, apiSecret))
                .map { it.nonZeroBalances }
                .compose(HttpErrorTransformer())
    }
}
